lib/logstash/inputs/redis.rb

```diff
@@ -186,15 +187,8 @@ def list_runner(output_queue)
       begin
         @redis ||= connect
         @list_method.call(@redis, output_queue)
-      rescue ::Redis::BaseError => e
-        info = { message: e.message, exception: e.class }
-        info[:backtrace] = e.backtrace if @logger.debug?
-        @logger.warn("Redis connection problem", info)
-        # Reset the redis variable to trigger reconnect
-        @redis = nil
-        # this sleep does not need to be stoppable as its
-        # in a while !stop? loop
-        sleep 1
+      rescue => e
+        retry if handle_error(e)
```

Comment:
Nitpick: I would expect a method called `handle_error` to return true if the error had been handled, but the way that we retry here indicates the opposite.